src/com/google/javascript/jscomp/CompilerOptions.java

```diff
@@ -2369,20 +2388,13 @@ public boolean isStrict() {
     /** Whether this is ECMAScript 5 or higher. */
     public boolean isEs5OrHigher() {
       Preconditions.checkState(this != NO_TRANSPILE);
-      return this != LanguageMode.ECMASCRIPT3;
+      return this.toInteger() >= 5;
     }
 
     /** Whether this is ECMAScript 6 or higher. */
     public boolean isEs6OrHigher() {
       Preconditions.checkState(this != NO_TRANSPILE);
-      switch (this) {
-        case ECMASCRIPT6:
-        case ECMASCRIPT6_STRICT:
-        case ECMASCRIPT6_TYPED:
-          return true;
-        default:
-          return false;
-      }
+      return this.toInteger() >= 6;
```

Comment:
return this.toInteger() == 6;

Since toInteger never returns > 6, I find the >= a bit confusing (even though it's more future-proof).